python/tvm/testing/utils.py

```diff
@@ -981,6 +982,38 @@ def _corstone300_compile_time_check():
 requires_vitis_ai = Feature("vitis_ai", "Vitis AI", cmake_flag="USE_VITIS_AI")
 
 
+def _arm_dot_supported():
+    arch = platform.machine()
+
+    if arch not in ["arm64", "aarch64"]:
+        return False
+
+    if sys.platform.startswith("darwin"):
+        cpu_info = subprocess.check_output("sysctl -a", shell=True).strip().decode()
+        for line in cpu_info.split("\n"):
+            if line.startswith("hw.optional.arm.FEAT_DotProd"):
+                return bool(int(line.split(":", 1)[1]))
+    elif sys.platform.startswith("linux"):
+        return True
+
+    return False
+
+
+def _has_vnni():
+    arch = platform.machine()
+    # Only linux is supported for now.
+    if arch == "x86_64" and sys.platform.startswith("linux"):
+        return "avx512_vnni" in open("/proc/cpuinfo", "r").read()
```

Comment:
There's a bunch of different extensions to vnni. Do we use any of those extensions? If so, we might need another check for them.

Reply:
From the output of cpuinfo, `avx512_vnni` is the only relevant extension. There are 8 bit and 16 bit input variants, and we only use the former.